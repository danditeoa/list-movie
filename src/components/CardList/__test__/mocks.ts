export const mockList = [
  {
    adult: false,
    backdrop_path: '/y5Z0WesTjvn59jP6yo459eUsbli.jpg',
    genre_ids: [27, 53],
    id: 663712,
    original_language: 'en',
    original_title: 'Terrifier 2',
    overview:
      "After being resurrected by a sinister entity, Art the Clown returns to Miles County where he must hunt down and destroy a teenage girl and her younger brother on Halloween night.  As the body count rises, the siblings fight to stay alive while uncovering the true nature of Art's evil intent.",
    popularity: 7866.782,
    poster_path: '/b6IRp6Pl2Fsq37r9jFhGoLtaqHm.jpg',
    release_date: '2022-10-06',
    title: 'Terrifier 2',
    video: false,
    vote_average: 7.1,
    vote_count: 445,
  },
  {
    adult: false,
    backdrop_path: '/bQXAqRx2Fgc46uCVWgoPz5L5Dtr.jpg',
    genre_ids: [28, 14, 878],
    id: 436270,
    original_language: 'en',
    original_title: 'Black Adam',
    overview:
      'Nearly 5,000 years after he was bestowed with the almighty powers of the Egyptian gods—and imprisoned just as quickly—Black Adam is freed from his earthly tomb, ready to unleash his unique form of justice on the modern world.',
    popularity: 4428.609,
    poster_path: '/3zXceNTtyj5FLjwQXuPvLYK5YYL.jpg',
    release_date: '2022-10-19',
    title: 'Black Adam',
    video: false,
    vote_average: 6.9,
    vote_count: 905,
  },
  {
    adult: false,
    backdrop_path: '/pGx6O6IwqADOsgmqWzPysmWnOyr.jpg',
    genre_ids: [28, 14],
    id: 732459,
    original_language: 'en',
    original_title: 'Blade of the 47 Ronin',
    overview:
      'In this sequel to "47 Ronin," a new class of warriors emerges among the Samurai clans to keep a sought-after sword from falling into the wrong hands.',
    popularity: 3821.739,
    poster_path: '/kjFDIlUCJkcpFxYKtE6OsGcAfQQ.jpg',
    release_date: '2022-10-25',
    title: 'Blade of the 47 Ronin',
    video: false,
    vote_average: 6.7,
    vote_count: 47,
  },
  {
    adult: false,
    backdrop_path: '/naNXYdBzTEb1KwOdi1RbBkM9Zv1.jpg',
    genre_ids: [27, 53],
    id: 420634,
    original_language: 'en',
    original_title: 'Terrifier',
    overview:
      'On Halloween night, a young woman finds herself as the obsession of a sadistic murderer known as Art the Clown.',
    popularity: 3031.419,
    poster_path: '/6PQqC4SbY910VvyVad6mvsboILU.jpg',
    release_date: '2016-10-15',
    title: 'Terrifier',
    video: false,
    vote_average: 6.6,
    vote_count: 889,
  },
  {
    adult: false,
    backdrop_path: '/b2FxWOxe9K7ZZ1uaPOze2RJ1ajq.jpg',
    genre_ids: [27, 28, 35],
    id: 675054,
    original_language: 'es',
    original_title: 'MexZombies',
    overview:
      'A group of teenagers must face a zombie apocalypse, and help reestablish order.',
    popularity: 2784.235,
    poster_path: '/pTxw4GFE3ZfzJfvuUsPyD5njAuI.jpg',
    release_date: '2022-10-26',
    title: 'MexZombies',
    video: false,
    vote_average: 7.3,
    vote_count: 49,
  },
  {
    adult: false,
    backdrop_path: '/1DBDwevWS8OhiT3wqqlW7KGPd6m.jpg',
    genre_ids: [53],
    id: 985939,
    original_language: 'en',
    original_title: 'Fall',
    overview:
      'For best friends Becky and Hunter, life is all about conquering fears and pushing limits. But after they climb 2,000 feet to the top of a remote, abandoned radio tower, they find themselves stranded with no way down. Now Becky and Hunter’s expert climbing skills will be put to the ultimate test as they desperately fight to survive the elements, a lack of supplies, and vertigo-inducing heights',
    popularity: 2350.808,
    poster_path: '/spCAxD99U1A6jsiePFoqdEcY0dG.jpg',
    release_date: '2022-08-11',
    title: 'Fall',
    video: false,
    vote_average: 7.3,
    vote_count: 1650,
  },
];
