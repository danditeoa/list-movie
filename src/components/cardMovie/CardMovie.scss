.card {
  display: flex;
  flex-direction: column;
  width: 300px;
  background-color: #fff;
  border-radius: 15px;
  -webkit-box-shadow: 0px -1px 20px 0px rgba(110, 106, 87, 1);
  -moz-box-shadow: 0px -1px 20px 0px rgba(110, 106, 87, 1);
  box-shadow: 0px -1px 20px 0px rgba(110, 106, 87, 1);
  cursor: pointer;
  background-color: #FAF9F6;
}

img {
  border-radius: 15px 15px 0 0;
}

h2 {
  font-size: 20px;
  font-weight: 600;
}