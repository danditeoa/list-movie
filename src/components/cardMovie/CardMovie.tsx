import React from 'react';
import { Movie } from '../../models/paginatedMovie.model';
import "./CardMovie.scss";

const CardMovie = ({movie}: any) => {
 
const imageURL = "https://image.tmdb.org/t/p/w300/";
  return (
    <div className="card">
        
            <img src={`${imageURL}${movie.backdrop_path}`} alt="image backdrop" />

        <h2>
          {movie.title}
        </h2>
        <p>
          language: {movie.original_language}
        </p>
    </div>
  );
};

export default CardMovie;