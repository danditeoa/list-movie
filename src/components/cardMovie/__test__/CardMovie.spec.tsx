import { render, screen } from '@testing-library/react';
import { mockList } from '../../CardList/__test__/mocks';
import CardMovie from '../CardMovie';

const movieMock = mockList[0];
describe('<CardMovie />', () => {
  it('matches snapshot', () => {
    const { container } = render(<CardMovie movie={movieMock} />);
    expect(container).toMatchSnapshot();
  });

  it('renders title movie', () => {
    render(<CardMovie movie={movieMock} />);
    expect(screen.getByText('Terrifier 2')).toBeInTheDocument();
  });
});
