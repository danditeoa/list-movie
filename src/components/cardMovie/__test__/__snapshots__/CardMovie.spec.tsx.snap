// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardMovie /> matches snapshot 1`] = `
<div>
  <div
    class="card-container"
  >
    <img
      alt="image backdrop"
      src="https://image.tmdb.org/t/p/w300//y5Z0WesTjvn59jP6yo459eUsbli.jpg"
    />
    <h2>
      Terrifier 2
    </h2>
    <p>
      <strong>
        2022
      </strong>
    </p>
  </div>
</div>
`;
